src/quicklist.c

```diff
@@ -46,7 +46,8 @@
 /* Optimization levels for size-based filling.
  * Note that the largest possible limit is 64k, so even if each record takes
  * just one byte, it still won't overflow the 16 bit count field. */
-static const size_t optimization_level[] = {4096, 8192, 16384, 32768, 65536};
+static const size_t optimization_level[] = {4096, 8192, 16384, 32768, 65535};
+static_assert(((1 << (QL_LIMIT_BITS)) - 1 >= 65535), "QL_LIMIT_BITS is too small");
```

Comment:
```suggestion
#define LIMIT_MAX ((1 << (QL_LIMIT_BITS-1))-1)
static_assert(LIMIT_MAX >= 65535), "QL_LIMIT_BITS is too small");
```

then we can remove this line: `
#define FILL_MAX ((1 << (QL_LIMIT_BITS-1))-1)`

Reply:
Should we define the LIMIT_MAX to be `#define LIMIT_MAX ((1 << (QL_LIMIT_BITS))-1)` instead of `#define LIMIT_MAX ((1 << (QL_LIMIT_BITS-1))-1)`? Otherwise, the static_assert can't be statisfied. And the `FILL_MAX` (now is `LIMIT_MAX`) will change from 32767 to 65535.

Reply:
yes, mistake, it should be `#define LIMIT_MAX ((1 << (QL_LIMIT_BITS))-1)`